contracts/cw20-merkle-airdrop/src/contract.rs

```diff
@@ -0,0 +1,487 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    attr, to_binary, Addr, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult, SubMsg,
+    Uint128, WasmMsg,
+};
+
+use crate::state::{Config, CLAIM, CONFIG, MERKLE_ROOT, STAGE};
+
+use crate::error::ContractError;
+use crate::msg::{
+    ConfigResponse, ExecuteMsg, InstantiateMsg, IsClaimedResponse, LatestStageResponse,
+    MerkleRootResponse, MigrateMsg, QueryMsg,
+};
+use cw20::Cw20ExecuteMsg;
+use cw_storage_plus::U8Key;
+use sha3::Digest;
+use std::convert::TryInto;
+
+// Version info, for migration info
+const _CONTRACT_NAME: &str = "crates.io:cw20-merkle-airdrop";
+const _CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> StdResult<Response> {
+    let config = Config {
+        owner: deps.api.addr_validate(&msg.owner)?,
+        cw20_token_address: deps.api.addr_validate(&msg.cw20_token_address)?,
+    };
+    CONFIG.save(deps.storage, &config)?;
+
+    let stage = 0;
+    STAGE.save(deps.storage, &stage)?;
+
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::UpdateConfig { owner } => execute_update_config(deps, env, info, owner),
+        ExecuteMsg::RegisterMerkleRoot { merkle_root } => {
+            execute_register_merkle_root(deps, env, info, merkle_root)
+        }
+        ExecuteMsg::Claim {
+            stage,
+            amount,
+            proof,
+        } => execute_claim(deps, env, info, stage, amount, proof),
+    }
+}
+
+pub fn execute_update_config(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    owner: Option<String>,
+) -> Result<Response, ContractError> {
+    // authorize owner
+    let cfg = CONFIG.load(deps.storage)?;
+    if info.sender != cfg.owner {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // validate owner change
+    let new_owner = owner.ok_or(ContractError::InvalidInput {})?;
+    let new_owner = deps.api.addr_validate(&new_owner)?;
+
+    CONFIG.update(deps.storage, |mut exists| -> StdResult<_> {
+        exists.owner = new_owner;
+        Ok(exists)
+    })?;
+
+    Ok(Response {
```

Comment:
This isn't on 0.16 yet is it?

`Ok(Response::new().add_attribute("action", "update_config"))`

is now a one-liner